src/multibody/parser/urdf-with-geometry.hxx

```diff
@@ -133,34 +139,37 @@ namespace se3
           throw std::invalid_argument(exception_message);
         }
 
-      } // if(link->inertial)
+      }
       
       BOOST_FOREACH(::urdf::LinkConstPtr child,link->child_links)
       {
-        parseTreeForGeom(child, model, geom_model, meshRootDir, rootJointAdded);
+        parseTreeForGeom(child, model, geom_model, package_dirs, rootJointAdded);
       }
+
     }
 
-    GeometryModel buildGeom(const Model & model,
-                            const std::string & filename,
-                            const std::string & meshRootDir)
+
+    inline GeometryModel buildGeom(const Model & model,
+                                  const std::string & filename,
+                                  const std::vector<std::string> & package_dirs,
+                                  const bool root_joint)
```

Comment:
How do you know then if you must search for a "root_joint" or not ? as there are two buildModel procedures